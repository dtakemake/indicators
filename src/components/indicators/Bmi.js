import React from 'react'

/*
* Индекс массы тела
* @param height сантиметры
* @param weight килограммы
* @return bmi float 00.00
*/

const Bmi = ({ height = 0, weight = 0 }) => {

  /* делаем проверку только на рост, 
  чтобы не делить на ноль и мучатсья с проблемами рассчетов */

  return  <h3>ИМТ - { (height && (weight / Math.pow((height / 100), 2)).toFixed(2)) || '0.00' }</h3>
}

export default React.memo(Bmi)