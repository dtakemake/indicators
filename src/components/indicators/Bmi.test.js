import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import Bmi from './Bmi'

describe('Bmi', () => {
 
  let props = {
    height : 175,
    weight : 80
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Bmi />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Bmi { ...props } />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('есть корректный снимок с корректными переменными', () => {
    const component = TestRenderer.create(<Bmi { ...props } />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('есть корректный снимок без props', () => {
    const component = TestRenderer.create(<Bmi />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})