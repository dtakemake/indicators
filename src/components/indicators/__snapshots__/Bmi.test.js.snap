// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bmi есть корректный снимок без props 1`] = `
<h3>
  ИМТ - 
  0.00
</h3>
`;

exports[`Bmi есть корректный снимок с корректными переменными 1`] = `
<h3>
  ИМТ - 
  26.12
</h3>
`;
